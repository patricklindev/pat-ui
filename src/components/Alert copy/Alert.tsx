import React, { FC, ReactNode } from 'react';
import { classNames } from '../../utils/classNames';
import Button from '../Button/Button';

export type AlertType = 'standard' | 'filled' | 'outlined';
export type AlertIcon = 'error' | 'warning' | 'info' | 'success';
export type AlertSuffix = 'text' | 'button' | 'closable';
export type AlertTextStyle = 'one-line' | 'one-line-bold' | 'two-line' | 'two-line-bold';

export interface AlertProps {
  /** set customized style */
  className?: string;
  /** set alert Type */
  atType: AlertType;
  /** set alert Icon */
  atIcon: AlertIcon;
  /** set alert Suffix */
  atSuffix?: AlertSuffix;
  /** set alert text */
  atTextStyle: AlertTextStyle;
  /** set action on bottun clicked */
  btnOnClick?:() => void;
}

export const Alert: FC<AlertProps> = (props) => {
  const {atIcon, atSuffix, atTextStyle, atType, children, className, ...rest} = props;

  console.log(props)
  
  let styleClasses = classNames('alert', {
    [`alert-${atIcon}`]: true,
    [`alert-${atSuffix}`]: true,
    [`alert-${atTextStyle}`]: true,
    [`alert-${atType}`]: true,
  });
  console.log(styleClasses);
  
  // add user's custom class
  if (className) {
    styleClasses += ' ' + className;
  }

  // render different component based on type
  let alertComponent: JSX.Element | undefined;
  alertComponent = (
    <div className={styleClasses} data-testid='stanrdard-element'>
      <div className="alert-message">{children}</div>
      <div className='alert-suffix'>
        {atSuffix === 'button' && (
          <Button className="alert-action">
            Button
          </Button>
        )}
      </div>
    </div>
  );
  
  return <>{alertComponent}</>;
}


Alert.defaultProps = {
  atType: 'standard',
  atSuffix: 'text'
}

export default Alert;
