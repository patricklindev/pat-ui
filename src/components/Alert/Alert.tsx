import React from 'react'

export default function Alert() {
  return (
    <div>Alert</div>
  )
}
