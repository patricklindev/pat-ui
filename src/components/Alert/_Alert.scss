.alert {
  position: relative;
  font-weight: $alert-font-weight;
  line-height: $alert-line-height;
  color: $body-color;
  white-space: nowrap;
  text-align: center;
  vertical-align: middle;
  border: $alert-border-width solid $gray-500;
  padding: $alert-padding-y $alert-padding-x;
  border-radius: $alert-border-radius;
  margin: 0.5rem;
}

.alert-sm {
  font-size: $alert-font-size-sm;
  padding: $alert-padding-y-sm $alert-padding-x-sm;
}

.alert-lg {
  font-size: $alert-font-size-lg;
  padding: $alert-padding-y-lg $alert-padding-x-lg;
}
.alert-default {
  @include alert-style(transparent, $body-color, $gray-500);
}
.alert-primary {
  @include alert-style(rgba($primary, 0.3), darken($primary, 10%), $primary);
}

.alert-secondary {
  @include alert-style(
    rgba($secondary, 0.3),
    darken($secondary, 10%),
    $secondary
  );
}

.alert-danger {
  @include alert-style(rgba($danger, 0.3), darken($danger, 10%), $danger);
}

.alert-warning {
  @include alert-style(rgba($warning, 0.3), darken($warning, 10%), $warning);
}

.alert-info {
  @include alert-style(rgba($info, 0.3), darken($info, 10%), $info);
}

.alert-success {
  @include alert-style(rgba($success, 0.3), darken($success, 10%), $success);
}
