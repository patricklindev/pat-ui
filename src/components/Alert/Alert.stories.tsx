import React from 'react';
import { action } from '@storybook/addon-actions';
import Alert from './Alert';

export default {
    title: 'Alert',
    component: Alert,
  };

export const DefaultAlert = () => (
    <Alert>Default Alert-Check it out</Alert>
)

export const AlertSizes = () => (
    <div>
      <Alert alertSize="sm">Small Default Alert-Check it out</Alert>
      <Alert>Default Alert-Check it out</Alert>
      <Alert alertSize="lg">Large Default Alert-Check it out</Alert>
    </div>
)

export const AlertTypes = () => (
    <div>
      <Alert alertType="default">Default Alert-Check it out</Alert>
      <Alert alertType="primary">Primary  Alert-Check it out</Alert>
      <Alert alertType="secondary">Secondary  Alert-Check it out</Alert>
      <Alert alertType="danger">Danger  Alert-Check it out</Alert>
      <Alert alertType="warning">Warning  Alert-Check it out</Alert>
      <Alert alertType="info">Info  Alert-Check it out</Alert>
      <Alert alertType="success">Success  Alert-Check it out</Alert>
    </div>
)