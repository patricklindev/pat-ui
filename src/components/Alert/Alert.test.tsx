import React from 'react';

import { render} from '@testing-library/react';
import Alert, {PatAlertProps} from './Alert';

describe('Alert', ()=> {
    it('should match snapshot', () => {
        const { asFragment } = render(<Alert> Snapshot Alert </Alert>);
        expect(asFragment()).toMatchSnapshot();
      });

      it('should render default alert', () => {
        const wrapper = render(<Alert>Default Alert</Alert>);
        const element = wrapper.queryByText('Default Alert') as HTMLElement;
        expect(element).toBeInTheDocument();
        expect(element.tagName).toBe('DIV');
        expect(element).toHaveClass('alert alert-default');
      });

      it('should render correct alert based on different props', () => {
        const alertPrimarySmallProps: PatAlertProps = {
          alertType: 'primary',
          alertSize: 'sm',
          className: 'test',
        };
        const alertPrimarySmallWrapper = render(
          <Alert {...alertPrimarySmallProps}>Primary Small Alert</Alert>
        );
        const alertPrimarySmallElement = alertPrimarySmallWrapper.queryByText(
          'Primary Small Alert'
        ) as HTMLElement;
        expect(alertPrimarySmallElement).toBeInTheDocument();
        expect(alertPrimarySmallElement.tagName).toBe('DIV');
        expect(alertPrimarySmallElement).toHaveClass('alert alert-primary alert-sm test');
      });
});