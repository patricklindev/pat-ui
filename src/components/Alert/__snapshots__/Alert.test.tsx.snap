// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-default"
  >
     Snapshot Alert 
  </div>
</DocumentFragment>
`;
